parsers/US_CA.py

```diff
@@ -77,30 +86,48 @@ def fetch_production(
         df = csv.copy().iloc[:-1]
     else:
         df = csv.copy()
-    df = df.rename(columns={**PRODUCTION_MODES_MAPPING, **STORAGE_MAPPING})
-    df = df.groupby(level=0, axis=1).sum()
 
     all_data_points = []
     for index, row in df.iterrows():
         production = {}
-        storage = {}
-
+        storage = {"hydro": 0.0, "battery": 0.0}
         row_datetime = target_datetime.replace(
             hour=int(row["Time"][:2]), minute=int(row["Time"][-2:])
         )
-        for mode in [mode for mode in PRODUCTION_MODES if mode in row]:
+        for mode in [
+            mode
+            for mode in PRODUCTION_MODES_MAPPING
+            if mode not in ["Small hydro", "Large Hydro"]
+        ]:
             production_value = float(row[mode])
             if production_value < 0 and (
-                mode in ["solar", "nuclear", "coal", "gas", "unknown", "biomass"]
+                mode
+                in [
+                    "Solar",
+                    "Wind",
+                    "Geothermal",
+                    "Biomass",
+                    "Biogas",
+                    "Coal",
+                    "Nuclear",
+                    "Natural Gas",
+                ]
             ):
                 logger.warn(
                     f" {mode} production for US_CA was reported as less than 0 and was clamped"
                 )
                 production_value = 0.0
 
-            production[mode] = production_value
+            production = add_production_to_dict(mode, production_value, production)
+
+        for mode in ["Small hydro", "Large Hydro"]:
+            production_value = float(row[mode])
+            if production_value < 0:
+                storage["hydro"] += production_value
```

Comment:
don't this need to be multiplied by -1 as well like the battery storage?
```suggestion
                storage["hydro"] += production_value * -1
```

Reply:
yes i was wondering the same thing but i think this makes sense